test/integration/tasks.test.js

```diff
@@ -249,4 +249,33 @@ describe('Tasks', function () {
         })
     })
   })
+
+  describe('GET /tasks/:username', function () {
+    it('Should return 200 when username is valid', function (done) {
+      chai
+        .request(app)
+        .get('/tasks/sagar')
+        .end((err, res) => {
+          if (err) { return done(err) }
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.equal('Tasks returned successfully!')
+          expect(res.body.tasks).to.be.a('array')
+          return done()
```

Comment:
Can we have stricter assertions on `res.body.tasks`?
What I mean is `expect(res.body.tasks).to.have.all.keys(...taskObjectKeys)`
Where `taskObjectKeys` is an array of the keys of the task object
You can refer this: https://github.com/Real-Dev-Squad/website-backend/blob/09c53d861f1133e9ab2bea3a1421bcec855f8716/test/integration/auction.test.js#L51

Reply:
@swarajpure I tried to add assertions for `res.body.tasks` but when I looked into the result of `res.body.tasks`. It outputs some different object which we don't have in any variable stored. This is what it outputs.
`[
  {
    id: '7f4MpynEfkksfRQkcTcG',
    percentCompleted: 10,
    endsOn: '<unix timestamp>',
    isNoteworthy: true,
    dependsOn: [ 'd12', 'd23' ],
    purpose: 'To Test mocha',
    lossRate: { dinero: 1 },
    title: 'Assigned task',
    type: 'Dev | Group',
    completionAward: { dinero: 3, neelam: 300 },
    links: [ 'test1' ],
    assignee: 'user1',
    featureUrl: '<testUrl>',
    startedOn: '<unix timestamp>',
    status: 'active',
    participants: [ 'sagar' ]
  },
  {
    id: 'NSbhS5GCQaI3ePrBVS4k',
    percentCompleted: 10,
    endsOn: 1234,
    isNoteworthy: true,
    lossRate: { dinero: 1 },
    title: 'new-title',
    type: 'feature',
    completionAward: { dinero: 3, neelam: 300 },
    startedOn: 4567,
    status: 'active',
    participants: [ 'sagar' ]
  },
  {
    id: 'eHeuYaOY3iFN3m49l4qg',
    percentCompleted: 10,
    endsOn: 1234,
    isNoteworthy: false,
    dependsOn: [ 'd12', 'd23' ],
    purpose: 'To Test mocha',
    lossRate: { dinero: 1 },
    title: 'Test task',
    type: 'group',
    completionAward: { dinero: 3, neelam: 300 },
    links: [ 'test1' ],
    featureUrl: '<testUrl>',
    startedOn: 54321,
    status: 'completed',
    participants: [ 'sagar' ]
  },
  {
    id: 'H848N7dbBT4l5SLfZf54',
    percentCompleted: 10,
    endsOn: 123,
    lossRate: { dinero: 1 },
    title: 'Test task - Create',
    type: 'feature',
    completionAward: { dinero: 3, neelam: 300 },
    createdBy: 'sagar',
    assignee: 'sagar',
    startedOn: 456,
    participants: [],
    status: 'completed'
  }
]
`

I think its adding elements to tasks dynamically when we execute previous test cases. Could you please help me understand what logic we can apply ?

Reply:
I get it. There are some optional fields.
Can we are least check for assignee field? Making sure it returns the `username` provided.

Reply:
Updated the code for this. Please review it.